src/Tools/dotnet-monitor/CollectionRules/Options/ConfigurationHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options
+{
+    internal static class ConfigurationHelper
+    {
+        public static bool TryConfigureOptions(Type optionsType, object options, IServiceProvider serviceProvider)
+        {
+            Type configureOptionsType = typeof(IConfigureOptions<>).MakeGenericType(optionsType);
+            MethodInfo configureOptionsConfigMethod = configureOptionsType.GetMethod(
+                name: nameof(IConfigureOptions<object>.Configure), 
+                bindingAttr: BindingFlags.Instance | BindingFlags.Public, 
+                binder: null,
+                types: new Type[] { optionsType }, 
+                modifiers: null);
+
+            bool foundConfigSvcs = false;
```

Comment:
I don't technically see anything wrong with doing it this way from a functional perspective, but it appears that `foundConfigSvcs` is really just a check to see whether `configOptsImpls` is empty . Couldn't we just return that check (e.g. `return configOptsImpls != null && configOptsImpls.Any();`) instead of redundantly setting `foundConfigSvcs` to true for each object?